analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CheckArgumentException.CSharp12.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+// Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/7714
+public class PrimaryConstructor(string s, string q)
+{
+    private readonly string _s = s ?? throw new ArgumentNullException(nameof(s)); // Compliant: s is a class parameter
+
+    public string GetQ()
+    {
+        return q ?? throw new ArgumentNullException(nameof(q)); // Compliant: q is a class parameter
+    }
+
+    public string GetQLambda() => q ?? throw new ArgumentNullException(nameof(q)); // Compliant: q is a class parameter
+}
+
+public class PrimaryConstructor2(string Prop)
+{
+    public class TestClass
+    {
+        public static string B { get; }
+
+        public string A { get; } = B ?? throw new ArgumentNullException(nameof(Prop)); // Noncompliant
```

Comment:
Can you add the message? It is not clear why this is Noncompliant. Is it because of the name mismatch?

Reply:
Please add a comment about this design decision.